src/main/java/bio/terra/service/filedata/FileService.java

```
@@ -187,10 +221,29 @@ FSItem lookupFSItem(String datasetId, String fileId, int depth) throws Interrupt
    */
   FSItem lookupFSItemByPath(String datasetId, String path, int depth) throws InterruptedException {
     Dataset dataset = datasetService.retrieveAvailable(UUID.fromString(datasetId));
-    return fileDao.retrieveByPath(dataset, path, depth);
+    CloudPlatformWrapper cloudPlatformWrapper =
+        CloudPlatformWrapper.of(dataset.getDatasetSummary().getStorageCloudPlatform());
+    if (cloudPlatformWrapper.isGcp()) {
+      return fileDao.retrieveByPath(dataset, path, depth);
+    } else {
+      BillingProfileModel billingProfileModel =
+          profileService.getProfileByIdNoCheck(dataset.getDefaultProfileId());
+      AzureStorageAccountResource storageAccountResource =
+          resourceService
+              .getStorageAccount(dataset, billingProfileModel)
+              .orElseThrow(
```

should we combine this logic (throw if not found) in the resource service?

---

I agree with this sentiment. No use repeating ourselves.